### DISCUSSION TITLE
Minimum browser requirements for push notifications?

### DISCUSSION THREAD

**Discussion by reinhart1010** [mastodon/mastodon#18472] [Q&A]
At https://github.com/webcompat/web-bugs/issues/103392 someone reported unable to activate push notifications from their instance's settings in Firefox for Android. I believe this is due to feature detection (instead of infamous Firefox/Chrome 100 user agent mayhem) instead. We have confirmed this issue on another, independent Mastodon instance as well.

Looking back at [MDN](https://developer.mozilla.org/en-US/docs/Web/API/notification) Firefox for Android does not support certain notification features (like [`actions`](https://developer.mozilla.org/en-US/docs/Web/API/Notification/actions)). Are there any unsupported features which Mastodon currently require to utilize web push notifications?

**Comment by ClearlyClaire** [comment]
The exact check for browser support is the following:
```javascript
const supportsPushNotifications = ('serviceWorker' in navigator && 'PushManager' in window && 'getKey' in PushSubscription.prototype);
```

I have tried with Fennec installed from F-Droid, and this does returns `true`, but then trying to register an actual push subscription errors out with an unhelpful `DOMException: Error retrieving push subscription.`. I am not sure what is causing this.

**Reply by ClearlyClaire** [reply]
Installing Firefox from the Google Play Store works just fine. I suppose this is simply that the Firefox build that you are using does not support web push at all. It seems Firefox for Android relies on the Google Play Services for push notifications.

**Reply by rneilson** [reply]
I'll second having this problem, Firefox installed from Google Play Store, version 106.1.0.

**Comment by EtzBetz** [comment]
I just noticed that I was unable to get any notifications **on my phone via the app**, because I had blocked notifications **on my browser on deskop**. I now did unblock notifications in the browser's ui and then could enable push notifications in the mastodon ui.

I searched for this for 2 years.. this can't be valid ux. But is this the right place or should I create a new issue?

**Reply by ClearlyClaire** [reply]
Please open a new issue
